File: src/main/java/com/koddy/server/auth/infrastructure/oauth/zoom/response/ZoomTokenResponse.java
package com.koddy.server.auth.infrastructure.oauth.zoom.response;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import com.koddy.server.auth.domain.model.oauth.OAuthTokenResponse;

@JsonNaming(value = PropertyNamingStrategies.SnakeCaseStrategy.class)
public record ZoomTokenResponse(
        String tokenType,
        String accessToken,
        String refreshToken,
        String scope,
        long expiresIn
) implements OAuthTokenResponse {
}

File: src/main/kotlin/com/koddy/server/auth/infrastructure/social/zoom/response/ZoomTokenResponse.kt
package com.koddy.server.auth.infrastructure.social.zoom.response

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import com.koddy.server.auth.domain.model.oauth.OAuthTokenResponse

@JsonNaming(value = PropertyNamingStrategies.SnakeCaseStrategy::class)
data class ZoomTokenResponse(
    val tokenType: String,
    val accessToken: String,
    val refreshToken: String?,
    val scope: String,
    val expiresIn: Long,
) : OAuthTokenResponse {
    override fun accessToken(): String = accessToken
}
